isis/src/newhorizons/apps/leisa2isis/main.cpp

```diff
@@ -1,307 +1,27 @@
-
-    //just set to Isis::Real
-    if (cao.propagatePixelType()) {
-      cao.setPixelType(Isis::Real); 
+  catch (...) {
+    for (auto grpIt = appLog.beginGroup(); grpIt!= appLog.endGroup(); grpIt++) {
+      Application::Log(*grpIt);
     }
-
-    outputCube = importFits.SetOutputCube(qualityFile.toString(), cao); 
-
-    importFits.Progress()->SetText("Preparing quality image for comparing against LEISA pixels");
-    importFits.StartProcess(processFunc);
-    importFits.ClearCubes();
-    importFits.Finalize();
-
-    //now we have the temp cube and want to use fx to add it to the DN cube
-    //fx f1=temp_quality.cub f2=temp_dn.cub to=output_name.cub equation="f1 + f2"
-    QString parameters;
-    parameters += " F1= " + outputFile.toString();
-    parameters += " F2= " + qualityFile.toString();
-    parameters += " TO= " + ui.GetFileName("TO");
-    parameters += " EQUATION=" + QString("\"f1+f2\"");
-    ProgramLauncher::RunIsisProgram("fx", parameters);
-
-    // Cleanup by removing temporary cubes
-    remove(qualityFile.toString().toStdString().c_str());
-    remove(outputFile.toString().toStdString().c_str()); 
+    throw;
   }
-
-  // Import the ERRORMAP image. It is the 6th image in the FITS file (i.e., 5th extension)
-  if (ui.WasEntered("ERRORMAP")) {
-    PvlGroup extensionLabel = importFits.fitsImageLabel(5);
-    importFits.SetOrganization(ProcessImport::BIL);
-    importFits.setProcessFileStructure(5);
-    Cube *output = importFits.SetOutputCube("ERRORMAP");
-
-    // Save the input FITS label in the Cube original labels
-    Pvl origLabel;
-    origLabel += extensionLabel;
-    OriginalLabel originals(origLabel);
-    output->write(originals);
-
-    // Convert the main image data
-    importFits.Progress()->SetText("Importing LEISA errormap image");
-    importFits.StartProcess();
-    importFits.ClearCubes();
+  
+  for (auto grpIt = appLog.beginGroup(); grpIt!= appLog.endGroup(); grpIt++) {
+    Application::Log(*grpIt);
   }
-
-  // Import the quality image. It is the 7th image in the FITS file (i.e., 6th extension)
-  if (ui.WasEntered("QUALITY")) {
-    PvlGroup extensionLabel = importFits.fitsImageLabel(6);
-    importFits.SetOrganization(ProcessImport::BIL);
-    importFits.setProcessFileStructure(6);
-    Cube *output = importFits.SetOutputCube("QUALITY");
-
-    // Save the input FITS label in the Cube original labels
-    Pvl origLabel;
-    origLabel += extensionLabel;
-    OriginalLabel originals(origLabel);
-    output->write(originals);
-
-    // Convert the main image data
-    importFits.Progress()->SetText("Importing LEISA quality image");
-    importFits.StartProcess();
-    importFits.ClearCubes();
-  }
-}
-
-void processFunc(Buffer &in){
-  for (int i = 0; i < in.size(); i++) {
-    if (((in[i] == 1) || (in[i] == 2)) || (in[i] == 8)) {
-      in[i] = Isis::NULL8;
-    } 
-    else {
-        in[i] = 0;
-    }
-  }
-  outputCube->write(in); 
-}
-
+}
```

Comment:
Silly I know, but no CR  at EOF. I'll add it and save.

Reply:
Thanks!